phpstan.neon.dist

```diff
@@ -0,0 +1,11 @@
+parameters:
+    level: max
+
+    paths:
+        - src/
+        - tests/
+
+    reportUnmatchedIgnoredErrors: false
+    ignoreErrors:
+        # ignore generic usage like `PromiseInterface<T>` until fixed upstream
+        - '/^PHPDoc .* contains generic type React\\Promise\\PromiseInterface<.+> but interface React\\Promise\\PromiseInterface is not generic\.$/'
```

Comment:
For reference, https://github.com/reactphp/promise/pull/223 and https://github.com/reactphp/promise/pull/188 will resolve this